component.go

```diff
@@ -211,113 +310,116 @@ func (l *LiveComponent) InvokeMethodInPath(path string, data map[string]string,
 	return nil
 }
 
-// Render ...
-func (l *LiveComponent) Render() (string, error) {
-	l.log(LogTrace, "Render", logEx{"name": l.Name})
-
-	if l.component == nil {
-		return "", errors.New("component nil")
-	}
-
-	s := bytes.NewBufferString("")
-
-	err := l.htmlTemplate.Execute(s, l.component)
-
-	if err != nil {
-		return "", err
-	}
-
-	return s.String(), nil
+func (l *LiveComponent) createUniqueName() string {
+	return l.Name + "_" + NewLiveId().GenerateSmall()
 }
 
-// LiveRender render a new version of the component, and detect
-// differences from the last render
-// and sets the "new old" version  of render
-func (l *LiveComponent) LiveRender() ([]OutMessage, error) {
-	newRender, err := l.Render()
-
-	if err != nil {
-		return nil, fmt.Errorf("render component: %w", err)
-	}
-
-	oms := make([]OutMessage, 0)
-	if len(l.rendered) > 0 {
-
-		changeInstructions, err := GetDiffFromRawHTML(l.rendered, newRender)
-
-		if err != nil {
-			l.log(LogPanic, "there is a error in diff", logEx{"error": err})
+// TODO: maybe nested components?
+func (l *LiveComponent) getChildrenComponents() []*LiveComponent {
+	components := make([]*LiveComponent, 0)
+	v := reflect.ValueOf(l.component).Elem()
+	for i := 0; i < v.NumField(); i++ {
+		if !v.Field(i).CanInterface() {
+			continue
 		}
 
-		for _, instruction := range changeInstructions {
-			oms = append(oms, OutMessage{
-				Name:        EventLiveDom,
-				Type:        strconv.Itoa(int(instruction.Type)),
-				Attr:        instruction.Attr,
-				ComponentId: instruction.componentId,
-				Content:     instruction.Content,
-				Element:     instruction.Element,
-			})
+		lc, ok := v.Field(i).Interface().(*LiveComponent)
+		if !ok {
+			continue
```

Comment:
Would a warning log be a good idea here? Is it valid to have a non-component as a child?